plugins/storage/volume/default/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/CloudStackPrimaryDataStoreLifeCycleImpl.java

```diff
@@ -141,6 +141,9 @@ public DataStore initialize(Map<String, Object> dsInfos) {
 
         URI uri = null;
         try {
+            if (url != null && url.startsWith("rbd://")) {
```

Comment:
in case the url scheme is not case sensitive, can use:

```suggestion
            if (url != null && url..toLowerCase().startsWith("rbd://")) {
```

(or)

```suggestion
            if (StringUtils.startsWithIgnoreCase(url, "rbd://")) {
```